src/Orleans.Core.Abstractions/GrainDirectory/IGrainDirectory.cs

```diff
@@ -58,5 +58,22 @@ public class GrainAddress
         /// Id of the specific Grain activation
         /// </summary>
         public string ActivationId { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            return obj is GrainAddress address &&
+                   this.SiloAddress == address.SiloAddress &&
+                   this.GrainId == address.GrainId &&
+                   this.ActivationId == address.ActivationId;
+        }
+
+        public override int GetHashCode()
+        {
+            var hashCode = 1043893337;
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(this.SiloAddress);
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(this.GrainId);
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(this.ActivationId);
+            return hashCode;
```

Comment:
We can pull in the Microsoft.Bcl.HashCode package and use `HashCode.Combine` here instead (but this is fine, too)

Reply:
I just want to avoid pulling too many dependencies on `Abstractions`...

@sergeybykov not sure what you are suggesting here. It's a `GetHashCode` generated from VS2019

Reply:
I'm fine with that.